wgsl/index.bs

```diff
@@ -3682,15 +3682,18 @@ function_decl
   : decoration_list* function_header body_statement
 
 function_type_decl
-  : type_decl
+  : decoration_list* type_decl
```

Comment:
should this be a single decoration instead of multiple ones?

Reply:
Maybe? But I guess making it a list makes the grammar more future proof. It's also consistent with all the other places that we allow decoration list even when there is only one valid decoration (e.g. `struct_decl`, `global_constant_decl ` etc).